nUpdate.UpdateInstaller/WindowsServiceHelper.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace nUpdate.UpdateInstaller
+{
+    public static class WindowsServiceHelper
+    {
+        private static bool? _IsRunningInServiceContext;
+
+        /// <summary>
+        /// Indicates if the application runs in service context
+        /// </summary>
+        public static bool IsRunningInServiceContext => _IsRunningInServiceContext ?? (_IsRunningInServiceContext = DetermineIfRunningInServiceContext()).Value; 
+
+
+
+        private static bool DetermineIfRunningInServiceContext()

```

Comment:
i would say we have one condition that we cant catch 
the application runs as service, has "allow service to interact with desktop" checked and! he has change the user ( login with ... )